climpred/toy.py

```diff
@@ -0,0 +1,259 @@
+
+    """
+    A = 1.0 / (1 + a * np.exp((t_opt - lead) / r)) - 1.0 / (1 + a * np.exp((t_opt) / r))
+    A[0] = 0
+    A = A / A[-1]
+    return A
+
+
+def create_noise(
+    lead=lead, ninit=3, nmember=3, noise_amplitude=noise_amplitude, ramp=ramp
+):
+    """Create gaussian noise.
+
+    Args:
+        lead (xr.DataArray): lead times to create synthetic data for.
+        ninit (type): number of initialzations. Defaults to 3.
+        nmember (type): number of members. Defaults to 3.
+        noise_amplitude (type): amplitude of the gaussian noise.
+        ramp (type): Ramping up function.
+
+    Returns:
+        xr.DataArray: ramped up noise with initializations and members
+
+    """
+    noise = (
+        noise_amplitude
+        * xr.DataArray(
+            np.random.rand(lead.size, ninit, nmember), dims=['lead', 'init', 'member']
+        )
+        - noise_amplitude / 2
+    )
+    noise['lead'] = lead.values
+    noise = noise * ramp(lead)
+    noise['member'] = np.arange(1, 1 + nmember)
+    noise['init'] = np.arange(1, 1 + ninit)
+    return noise
+
+
+def signal(
+    lead=lead, signal_amplitude=signal_amplitude, signal_P=signal_P, lead_offset=0
+):
+    """The low-frequency signal to be predicted.
+
+    Args:
+        lead (xr.DataArray): lead times to create synthetic data for.
+        signal_amplitude (float): amplitude of the signal.
+        signal_P (float): period of the signal.
+        lead_offset (float): phase of the signal. Defaults to 0.
+
+    Returns:
+        xr.DataArray: signal to be predicted
+
+    """
+    return signal_amplitude * xr.DataArray(
+        np.sin((lead - lead_offset) * np.pi * 2 / signal_P)
+    )
+
+
+def create_initialized(
+    lead=lead,
+    ninit=10,
+    nmember=10,
+    signal_amplitude=signal_amplitude,
+    noise_amplitude=noise_amplitude,
+    signal_P=signal_P,
+    ramp=ramp,
+):
+    """Create initialized ensemble.
+
+    Args:
+        lead (xr.DataArray): lead times to create synthetic data for.
+        ninit (type): number of initialzations. Defaults to 3.
+        nmember (type): number of members. Defaults to 3.
+        noise_amplitude (type): amplitude of the gaussian noise.
+        ramp (np.array): Ramping up function.
+        signal_amplitude (float): amplitude of the signal.
+        signal_P (float): period of the signal.
+
+    Returns:
+        xr.DataArray: initialized ensemble with potentially predictable variation.
+
+    """
+    # span range of initial conditions
+    init_range = 2 * signal_P
+    to = xr.DataArray(np.random.rand(ninit) * init_range - init_range, dims='init')
+    # create initialized
+    init = signal(
+        lead=lead, lead_offset=to, signal_amplitude=signal_amplitude, signal_P=signal_P
+    ) + create_noise(
+        lead=lead,
+        ninit=ninit,
+        nmember=nmember,
+        noise_amplitude=noise_amplitude,
+        ramp=ramp,
+    )
+    return init
+
+
+def run_skill_for_ensemble(
+    lead=lead,
```

Comment:
I don't know if it's good practice to set this here. Perhaps make it `lead=None` and then if `lead=None`, set lead to the default lead from the header of this script.